src/io/flutter/run/bazel/BazelFields.java

```diff
@@ -274,10 +278,19 @@ GeneralCommandLine getLaunchCommand(@NotNull Project project,
       commandLine.addParameter(additionalArgsTokenizer.nextToken());
     }
 
+    final String enableBazelHotRestartParam = "--enable-google3-hot-reload";
     final String disableBazelHotRestartParam = "--no-enable-google3-hot-reload";
-    if (!FlutterSettings.getInstance().isEnableBazelHotRestart() &&
-        !StringUtil.notNullize(additionalArgs).contains(disableBazelHotRestartParam)) {
-      commandLine.addParameter(disableBazelHotRestartParam);
+    final boolean hasEnabledArg = StringUtil.notNullize(additionalArgs).contains(enableBazelHotRestartParam);
+    final boolean hasDisabledArg = StringUtil.notNullize(additionalArgs).contains(disableBazelHotRestartParam);
+    if (!FlutterSettings.getInstance().isEnableBazelHotRestart() && hasDisabledArg) {
+      final Notification notification = new Notification(
+        FlutterMessages.FLUTTER_NOTIFICATION_GROUP_ID,
+        "Google3 hot restart is disabled by default",
```

Comment:
Nit: this makes it sound like hot restart is disabled rather than just "Google3 specific hot restart"